bin/git-magic

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+cd "$(git rev-parse --show-toplevel)"
+
+# Look for --all option
+for a in "$@"; do 
+  if [[ "$a" == "-a" || "$a" == "--all" ]]; then
+
+  	# Check if there is no changes to stage
+	if [[ -z $(git status --porcelain) ]]; then
+		break
+	fi
+	
+	# Get confirmation from user
+	git status
+	echo "Everything will be added"
+	read -p "Press enter to continue"
+
+	# Restore staging area so that, for example, 
+	# add and modify dont be seperate entries in status
+	git restore --staged .
+	git add .
+	break
+  fi
+done
+
+# Commit with generated message
+git commit --no-edit "$@" -m "$(git status --porcelain -uno)"
```

Comment:
Better to parse other options out instead of passing them directly

Reply:
```suggestion
ALL=false
ARGS=()
while [[ $# -gt 0 ]]; do
  key="$1"
  
  case $key in
    -F|-m)
      ARGS+=("$1")
      ARGS+=("$2")
      shift # past argument
      shift # past value
      ;;
    --file=*|--message=*|-e|--edit|--amend)
      ARGS+=("$1")
      shift # past argument
      ;;
    -a|--all)
      ALL=true
    *)    # unknown option
      echo "Unknown option $1"
      exit 1
      ;;
  esac
done

set -- "${ARGS[@]}" # restore positional parameters

```

Quoted "$@" passes arguments safely. Is it ok if I just filter compatible options that will be passed to `git commit`command?